packages/core/strapi/lib/services/entity-service/validation/relations.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const { ApplicationError } = require('@strapi/utils').errors;
+const { castArray } = require('lodash/fp');
+const { isEmpty, uniqBy } = require('lodash');
+
+/**
+ * Check that all the relations of an entity exist
+ */
+module.exports = async (data, uid, db) => {
+  const relationsMap = new Map();
+
+  const relationCheck = (data, uid) => {
+    const model = strapi.getModel(uid);
+    if (isEmpty(model)) return;
+
+    Object.keys(model.attributes).forEach((attributeName) => {
+      const attribute = model.attributes[attributeName];
+      const value = data[attributeName];
+      if (!value) {
+        return;
+      }
+      switch (attribute.type) {
+        case 'relation': {
+          if (!attribute.target) {
+            break;
+          }
+
+          // If the attribute type is a relation keep track of all
+          // associations being made with relations. These will later be checked
+          // against the DB to confirm they exist
+          let directValue = [];
+          if (Array.isArray(value)) {
+            directValue = value.map((v) => ({
+              id: v,
+            }));
+          }
+          relationsMap.set(attribute.target, [
+            ...(relationsMap.get(attribute.target) || []),
+            ...(value.connect || []),
+            ...(value.set || []),
+            ...directValue,
+          ]);
```

Comment:
In order to avoid recreating an array each time, what do you think of using an object for `relationsMap` and to use it like that:
```js
relationsMap[attribute.target] = relationsMap[attribute.target] || [];
relationsMap[attribute.target].push(...(value.connect || value.set || directValue))
```
?